gprofiler/metadata/py_module_version.py

```diff
@@ -0,0 +1,274 @@
+    site-packages directory.
+
+    :param entry: Path parts of the installed-files.txt entry.
+    :param info: Path parts of the egg-info directory relative to package root.
+    :returns: The converted entry.
+
+    For best compatibility with symlinks, this does not use ``abspath()`` or
+    ``Path.resolve()``, but tries to work with path parts:
+
+    1. While ``entry`` starts with ``..``, remove the equal amounts of parts
+       from ``info``; if ``info`` is empty, start appending ``..`` instead.
+    2. Join the two directly.
+    """
+    while entry and entry[0] == "..":
+        if not info or info[-1] == "..":
+            info += ("..",)
+        else:
+            info = info[:-1]
+        entry = entry[1:]
+    return str(pathlib.Path(*info, *entry))
+
+
+def _files_from_record(dist: pkg_resources.Distribution) -> Optional[Iterator[str]]:
+    """Based on _files_from_record in pip._internal.commands.show.search_packages_info"""
+    try:
+        text = dist.get_metadata("RECORD")
+    except FileNotFoundError:
+        return None
+    # This extra Path-str cast normalizes entries.
+    return (str(pathlib.Path(row[0])) for row in csv.reader(text.splitlines()))
+
+
+def _files_from_legacy(dist: pkg_resources.Distribution) -> Optional[Iterator[str]]:
+    """Based on _files_from_legacy in pip._internal.commands.show.search_packages_info"""
+    try:
+        text = dist.get_metadata("installed-files.txt")
+    except FileNotFoundError:
+        return None
+    paths = (p for p in text.splitlines(keepends=False) if p)
+    root = dist.location
+    info = dist.egg_info
+    if root is None or info is None:
+        return paths
+    try:
+        info_rel = pathlib.Path(info).relative_to(root)
+    except ValueError:  # info is not relative to root.
+        return paths
+    if not info_rel.parts:  # info *is* root.
+        return paths
+    return (_convert_legacy_entry(pathlib.Path(p).parts, info_rel.parts) for p in paths)
+
+
+def _get_package_name(dist: pkg_resources.Distribution) -> Optional[str]:
+    """Based on pip._internal.metadata.base.BaseDistribution.raw_name"""
+    metadata = _get_metadata(dist)
+    if metadata:
+        # The metadata should NEVER be missing the Name: key, but if it somehow
+        # does, fall back to the known canonical name.
+        return metadata.get("Name", dist.project_name)
+    return None
+
+
+# Matches the path to libpython as it may appear in /proc/[pid]/maps, e.g. " /usr/local/lib/libpython3.9.so.1.0"
+_LIBPYTHON_MAPS_PATTERN = re.compile(r"(?<=\s)/\S*libpython\S*\.so(\.\S+)?\Z")
+
+
+def _get_libpython_path(pid: int) -> Optional[str]:
+    try:
+        f = open(f"/proc/{pid}/maps")
+    except OSError:
+        return None
+
+    with f:
+        for line in f.readlines():
+            match = _LIBPYTHON_MAPS_PATTERN.search(line.strip())
+            if match is not None:
+                return match.group()
+    return None
+
```

Comment:
Can just use psutil `memory_maps`, as @Jongy did in https://github.com/Granulate/gprofiler/blob/719adaf7d9ab027fa2c705622420900e0c503612/gprofiler/profilers/java.py#L242